src/meta_data.c

```diff
@@ -18,37 +22,147 @@
         return -1;                                                  \
     } while (0)
 
-static int get_serial(const struct serial_number *sn)
+static int obtain_mac_from_iface(const char *iface, uint8_t *dst_mac)
 {
-    return 0;
+    int fd;
+    struct ifreq ifr;
+    unsigned char *mac;
+
+    fd = socket(AF_INET, SOCK_DGRAM, 0);
+    ifr.ifr_addr.sa_family = AF_INET;
+    strncpy(ifr.ifr_name , iface , IFNAMSIZ-1);
+    ioctl(fd, SIOCGIFHWADDR, &ifr);
+    close(fd);
+
+    mac = (unsigned char *)ifr.ifr_hwaddr.sa_data;
+    memcpy(dst_mac, mac, 6);
+
+	return 0;
 }
 
-static int get_mac_addr(const uint8_t *mac)
+/**
+ * Parse the path and obtain the PCI BDF from it
+ */
+static uint32_t get_bdf_from_path(const char *link_path)
+{
+    char *bdf;
+    size_t bdf_str_len;
+    char *colon1, *colon2, *dot, *slash1, *slash2, *null;
+    uint32_t ret = 0UL;
+    int x,y,z;
+
+    colon1 = strrchr(link_path, ':');
+    slash1 = colon1;
+    while (*slash1 && *slash1 != '/')
+        slash1--;
+
+    slash2 = strchr(colon1, '/');
+    bdf_str_len =  slash2 - slash1 - 1;
+    bdf = strndup(slash1 + 1, bdf_str_len);
+    bdf[bdf_str_len] = '\0';
+
+    colon1 = strchr(bdf, ':');
```

Comment:
Maybe do `colon1 += 1` as is later done with `colon2` for consistency.

Reply:
ok